shell/server/grain-server.js

```diff
@@ -273,12 +273,75 @@ Meteor.publish("grainSize", function (grainId) {
   });
 });
 
+const GRAIN_DELETION_MS = 1000 * 60 * 60 * 24 * 30; // thirty days
+SandstormDb.periodicCleanup(86400000, () => {
+  const trashExpiration = new Date(Date.now() - GRAIN_DELETION_MS);
+  Grains.find({ trashed: { $lt: trashExpiration } }).forEach((grain) => {
+    Grains.remove({ _id: grain._id });
```

Comment:
Good catch! Also, the clearing of `ApiTokens` should be outside of this iteration. Fixed.